src/Fluid/Flu_FixUp_Restrict.cpp

```diff
@@ -299,11 +299,19 @@ void Flu_FixUp_Restrict( const int FaLv, const int SonFluSg, const int FaFluSg,
 //    pressure, total energy density, and the dual-energy variable
 #     if ( MODEL == HYDRO )
 //    apply this correction only when preparing all fluid variables or magnetic field
+#     ifdef DEDT_NU
+#     ifdef MHD
+      if (  ( TVarCC & _TOTAL ) == ( _TOTAL - _DEDT_NU )  ||  ResMag  )
+#     else
+      if (  ( TVarCC & _TOTAL ) == ( _TOTAL - _DEDT_NU )  )
```

Comment:
@hfhsieh
- Strictly speaking, we should replace `TVarCC & _TOTAL` with `TVarCC & (_TOTAL-_DEDT_NU)` so that it also works when `TVarCC` includes `_DEDT_NU`.
- Furthermore, we actually set `const bool ResMag = false;` when disabling `MHD`. Perhaps we can also define `const long _DEDT_NU = 0;` (only inside this function) if it is not defined. By doing so, we can combine lines 302 to 314 into the following single line.
```
if (  ( TVarCC & (_TOTAL-_DEDT_NU) ) == ( _TOTAL - _DEDT_NU )  ||  ResMag  )
```

What do you think?

Reply:
@hyschive Agreed. It is more clear if we combine these condition. I think we can also define `const long ResVar = _TOTAL - _DEDT_NU;` in `Flu_FixUp_Restrict()` and rewrite this condition as 

~~~ c
if (  ( TVarCC & ResVar ) == ResVar  ||  ResMag  )
~~~

What do you think?